src/utils/py/daos_api.py

```diff
@@ -1629,6 +1629,82 @@ def get_attr(self, attr_names, coh=None, cb_func=None):
 
         return results
 
+class DaosSnapshot(object):
+    """ A python object that can represent a DAOS snapshot. We do not save the
+    coh in the snapshot since it is different each time the container is opened.
+    """
+    def __init__(self, context, name=None):
+        """The epoch is represented as a Python integer so when sending it to
+        libdaos we know to always convert it to a ctype.
+        """
+        self.context = context
+        self.name = name # currently unused
+        self.epoch = 0
+
+    def create(self, coh, epoch):
+        """ Send a snapshot creation request and store the info in the
+        DaosSnapshot object.
+        coh     --ctype.u_long handle on an open container
+        epoch   --the epoch number of the obj to be snapshotted
+        """
+        func = self.context.get_function('create-snap')
+        epoch = ctypes.c_uint64(epoch)
+        retcode = func(coh, ctypes.byref(epoch), None, None)
+        self.epoch = epoch.value
+        if retcode != 0:
+            raise DaosApiError("Snapshot create returned non-zero. RC: {0}"
+                               .format(retcode))
+
+    def list(self, coh):
+        """ List this snapshot and check that its epoch is the same as the once
+        originally created.
```

Comment:
It's not actually checking the epoch (which I don't think it should so just mentioning the comment needs updating)

Reply:
Right. I need to change this code.

Reply:
Software design wise the function should be true to its name and generally should do one thing without any unexpected side effects.  This function doesn't do what I would expect a function named list would do and I wouldn't expect it to be checking anything.  So for now okay, but plan on fixing this.